### PR TITLE
Introduced a few more extension methods to Path, which are clearer th…

### PR DESCRIPTION
**Pull request by shamsasari**
…an the static methods in Files.

Also migrated code away from the old File API.

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/internal/PathUtils.kt [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+/**
+ * Allows you to write code like: Paths.get("someDir") / "subdir" / "filename" but using the Paths API to avoid platform
+ * separator problems.
+ * @see Path.resolve
+ */
+operator fun Path.div(other: String): Path = resolve(other)
+
+/**
+ * @see Paths.get
```
Comment (tlil) [inline core/src/main/kotlin/net/corda/core/internal/PathUtils.kt]: Perhaps provide a short description as well? Ala `Path.div`

File: core/src/main/kotlin/net/corda/core/internal/PathUtils.kt [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    return targetFile
+}
+
+/** @see Files.copy */
+fun Path.copyTo(target: Path, vararg options: CopyOption): Path = Files.copy(this, target, *options)
+
+/** @see Files.move */
+fun Path.moveTo(target: Path, vararg options: CopyOption): Path = Files.move(this, target, *options)
+
+fun Path.copyTo(out: OutputStream): Long = Files.copy(this, out)
```
Comment (tlil) [inline core/src/main/kotlin/net/corda/core/internal/PathUtils.kt]: A few missing KDoc's in this file. #nitpick

File: core/src/test/kotlin/net/corda/core/internal/PathUtilsTest.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        dir.deleteRecursively()
+        assertThat(dir).doesNotExist()
+    }
+
+    @Test
+    fun `deleteRecursively - nested single file`() {
+        val dir = (tempFolder.root.toPath() / "dir1" / "dir2").createDirectories()
+        (dir / "file").createFile()
+        dir.deleteRecursively()
+        assertThat(dir).doesNotExist()
```
Comment (tlil) [inline core/src/test/kotlin/net/corda/core/internal/PathUtilsTest.kt]: Do you want to check that `dir` (`tempFolder / "dir1" / "dir2"`) doesn't exist here? Or `tempFolder / "dir1"`?

File: experimental/behave/src/main/kotlin/net/corda/behave/Utilities.kt [hunk 4]
```diff
@@ -3,6 +3,8 @@ package net.corda.behave
 import java.time.Duration
 import java.util.concurrent.CountDownLatch
 
+// TODO Most of these are available in corda core
```
Comment (tlil) [inline experimental/behave/src/main/kotlin/net/corda/behave/Utilities.kt]: :+1: Will address in follow-up PR

File: experimental/behave/src/main/kotlin/net/corda/behave/logging/LogUtilities.kt [hunk 5]
```diff
@@ -3,5 +3,7 @@ package net.corda.behave.logging
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 
+// TODO Already available in corda core
```
Comment (tlil) [inline experimental/behave/src/main/kotlin/net/corda/behave/logging/LogUtilities.kt]: :+1: Will address in follow-up PR

File: experimental/behave/src/main/kotlin/net/corda/behave/file/FileUtilities.kt [hunk 6]
```diff
@@ -1,14 +1,11 @@
 package net.corda.behave.file
 
-import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
 
-val currentDirectory: File
-    get() = File(System.getProperty("user.dir"))
+val currentDirectory: Path
```
Comment (tlil) [inline experimental/behave/src/main/kotlin/net/corda/behave/file/FileUtilities.kt]: Would this be useful to have in `PathUtils.kt`? If so, I can copy that across in my follow-up PR

File: experimental/behave/src/main/kotlin/net/corda/behave/node/Distribution.kt [hunk 7]
```diff
@@ -37,42 +39,39 @@ class Distribution private constructor(
...
+            cordaJar.parent.createDirectories()
+            url.openStream().use { it.copyTo(cordaJar) }
+        } catch (e: Exception) {
+            if ("HTTP response code: 401" in e.message!!) {
+                log.warn("CORDA_ARTIFACTORY_USERNAME ${System.getenv("CORDA_ARTIFACTORY_USERNAME")}")
+                log.warn("CORDA_ARTIFACTORY_PASSWORD ${System.getenv("CORDA_ARTIFACTORY_PASSWORD")}")
+                throw Exception("Incorrect Artifactory permission. Please set CORDA_ARTIFACTORY_USERNAME and CORDA_ARTIFACTORY_PASSWORD environment variables correctly.")
             } else {
-                throw Exception("File not found $cordaJar")
+                throw Exception("Invalid Corda version $version", e)
```
Comment (tlil) [inline experimental/behave/src/main/kotlin/net/corda/behave/node/Distribution.kt]: This exception message might be a bit misleading. Perhaps just re-throw `e`?

File: core/src/test/kotlin/net/corda/core/internal/PathUtilsTest.kt (hunk 3 above)
Comment (shamsasari) [reply core/src/test/kotlin/net/corda/core/internal/PathUtilsTest.kt]: Ah good catch, I should be deleting "dir1"!

File: experimental/behave/src/main/kotlin/net/corda/behave/file/FileUtilities.kt (hunk 6 above)
Comment (shamsasari) [reply experimental/behave/src/main/kotlin/net/corda/behave/file/FileUtilities.kt]: Yeah, makes sense.

File: experimental/behave/src/main/kotlin/net/corda/behave/node/Distribution.kt (hunk 7 above)
Comment (shamsasari) [reply experimental/behave/src/main/kotlin/net/corda/behave/node/Distribution.kt]: Sure

**Comment by shamsasari** [comment]
Comments addressed.

**Comment by tlil** [comment]
Great - looks good to me now

**Pull request merged**